zcash_client_backend/src/data_api.rs

```diff
@@ -1226,9 +1229,45 @@ pub trait WalletTest: WalletRead {
         &self,
         _txid: &TxId,
         _protocol: ShieldedProtocol,
-    ) -> Result<Vec<NoteId>, Self::Error> {
+    ) -> Result<Vec<NoteId>, <Self as WalletRead>::Error> {
         Ok(vec![])
     }
+
+    #[allow(clippy::type_complexity)]
+    fn get_confirmed_sends(
+        &self,
+        txid: &TxId,
+    ) -> Result<Vec<(u64, Option<String>, Option<String>, Option<u32>)>, <Self as WalletRead>::Error>;
```

Comment:
Could the inner result type here be a struct? I have no idea what this tuple means.

Reply:
Yeah, that would be nicer. I pulled this API as-is from #1533. We also need to add documentation etc. to the new APIs in this PR, which includes All Of The Tests (but I'd spent long enough on this PR).

Reply:
Opened #1544.